rebasehelper/checkers/abipkgdiff_tool.py

```diff
@@ -112,17 +119,80 @@ def run_check(cls, results_dir, **kwargs):
             command.append(pkg)
             command.append(new_pkg)
             logger.debug('Package name for ABI comparison %s', old_name)
-            output = os.path.join(cls.results_dir, results_dir, old_name + '-' + cls.log_name)
+            output = os.path.join(cls.results_dir, old_name + '.txt')
             try:
                 ret_code = ProcessHelper.run_subprocess(command, output_file=output)
             except OSError:
                 raise CheckerNotFoundError("Checker '%s' was not found or installed." % cls.CMD)
 
             if int(ret_code) & settings.ABIDIFF_ERROR and int(ret_code) & settings.ABIDIFF_USAGE_ERROR:
                 raise RebaseHelperError('Execution of %s failed.\nCommand line is: %s' % (cls.CMD, cmd))
-            if int(ret_code) == 0:
-                text = 'ABI of the compared binaries in package %s are equal.' % old_name
-            else:
-                text = 'ABI of the compared binaries in package %s are not equal.' % old_name
-            reports[output] = text
-        return reports
+            reports[old_name] = int(ret_code)
+        return {'packages': cls.parse_abi_logs(reports), 'abi_changes': cls.abi_changes, 'path': cls.results_dir}
+
+    @classmethod
+    def parse_abi_logs(cls, reports):
+        """
+        Parse abipkgdiff logs
+        :param reports: dictionary with paths to the logs of the produced (sub)packages
+        :return: returns packages names with its abipkgdiff changes
+        """
+        def parse_changes(chgs):
+            """
+            Parses each line with abipkgdiff output info
+            :param chgs: lines of changes
+            :return: dictionary of list of changes
+            """
+            ret_changes = {}
+            for line in chgs:
+                # Find all numbers of added/changed/deleted and get rid of '0' changes
+                m = re.search(r'(^[\w\s]+:)'
+                              r'.{1}([\d]+ Removed[\s\w\d()]*)'
+                              r'.{2}([\d]+ Changed[\s\w\d()]*)'
+                              r'.{2}([\d]+ Added[\s\w\d()]*)',
+                              line)
```

Comment:
The regexp is still far too generic, but for no good reason - if something changes substantially in the output of abipkgdiff, it will still fail to parse it.

Reply:
And you are ignoring filtered-out changes, which is IMHO quite important information.

Reply:
And it would be nice to add a comment with what such abipkgdiff output looks like, so we have some reference if it changes in the future.

Reply:
You are right, except the second comment, as the regex would match anything except `,` leading to storing the "filtered" counts to the same category(Removed/Changed/Added)